frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -77,12 +89,12 @@ export const getPageHeading = (buildStrategy: string, buildType?: string): strin
   switch (buildStrategy) {
     case BuildStrategyType.Source:
       return buildType === BuildSourceType.Binary
-        ? CreateApplicationFlow.JarUpload
-        : CreateApplicationFlow.Git;
+        ? i18n.t(EditFlowPageTitle[CreateApplicationFlow.JarUpload])
+        : i18n.t(EditFlowPageTitle[CreateApplicationFlow.Git]);
     case BuildStrategyType.Docker:
-      return CreateApplicationFlow.Dockerfile;
+      return i18n.t(EditFlowPageTitle[CreateApplicationFlow.Dockerfile]);
     default:
-      return CreateApplicationFlow.Container;
+      return i18n.t(EditFlowPageTitle[CreateApplicationFlow.Container]);
```

Comment:
I think the problem starts here with the name `getPageHeading`. This util method should still return an enum.

The translation should happen when rendering this data. That the value of the enum contains already the heading doesn't make this straight forward, but we should keep this for the moment.

(btw: IMHO the return type of this function should be `CreateApplicationFlow` instead of a string.)

```tsx
export const getApplicationFlow = (
  buildStrategy: string,
  buildType?: string,
): CreateApplicationFlow => {
  switch (buildStrategy) {
    case BuildStrategyType.Source:
      return buildType === BuildSourceType.Binary
        ? CreateApplicationFlow.JarUpload
        : CreateApplicationFlow.Git;
    case BuildStrategyType.Docker:
      return CreateApplicationFlow.Dockerfile;
    default:
      return CreateApplicationFlow.Container;
  }
};
```

Reply:
Fixed it !